examples/pipeline_tacotron2/text/text_preprocessing.py

```diff
@@ -63,23 +64,102 @@
 _letters = 'abcdefghijklmnopqrstuvwxyz'
 
 symbols = [_pad] + list(_special) + list(_punctuation) + list(_letters)
-_symbol_to_id = {s: i for i, s in enumerate(symbols)}
+_phonemizer = None
 
 
-def text_to_sequence(sent: str) -> List[int]:
+available_symbol_set = set(["english_characters", "english_phonemes"])
+available_phonemizers = set(["DeepPhonemizer"])
+
+
+def get_symbol_list(symbol_list: str = "english_characters",
+                    cmudict_root: Optional[str] = "./") -> List[str]:
+    if symbol_list == "english_characters":
+        return [_pad] + list(_special) + list(_punctuation) + list(_letters)
+    elif symbol_list == "english_phonemes":
+        return [_pad] + list(_special) + list(_punctuation) + CMUDict(cmudict_root).symbols
+    else:
+        raise ValueError(f"The `symbol_list` {symbol_list} is not supported."
+                         f"Supported `symbol_list` includes {available_symbol_set}.")
+
+
+def word_to_phonemes(sent: str, phonemizer: str, checkpoint: str) -> List[str]:
+    if phonemizer == "DeepPhonemizer":
+        from dp.phonemizer import Phonemizer
+        global _phonemizer
+        _other_symbols = ''.join(list(_special) + list(_punctuation))
+        _phone_symbols_re = r'(\[[A-Z]+?\]|' + '[' + _other_symbols + '])'  # [\[([A-Z]+?)\]|[-!'(),.:;? ]]
+
+        if _phonemizer is None:
+            # using a global variable so that we don't have to relode checkpoint
+            # everytime this function is called
+            _phonemizer = Phonemizer.from_checkpoint(checkpoint)
+
+        # Example:
+        # sent = "hello world!"
+        # '[HH][AH][L][OW] [W][ER][L][D]!'
+        sent = _phonemizer(sent, lang='en_us')
+
+        # ['[HH]', '[AH]', '[L]', '[OW]', ' ', '[W]', '[ER]', '[L]', '[D]', '!']
+        ret = re.findall(_phone_symbols_re, sent)
+
+        # ['HH', 'AH', 'L', 'OW', ' ', 'W', 'ER', 'L', 'D', '!']
+        ret = [r.replace("[", "").replace("]", "") for r in ret]
+
+        return ret
+    else:
+        raise ValueError(f"The `phonemizer` {phonemizer} is not supported."
+                         "Supported `symbol_list` includes `'DeepPhonemizer'`.")
+
+
+def text_to_sequence(sent: Iterable,
+                     symbol_list: Union[str, List[str]] = "english_characters",
+                     phonemizer: Optional[str] = "DeepPhonemizer",
+                     checkpoint: Optional[str] = "./en_us_cmudict_forward.pt",
+                     cmudict_root: Optional[str] = "./") -> List[int]:
     r'''Converts a string of text to a sequence of IDs corresponding to the symbols in the text.
 
-      Args:
-        sent (str): The input sentence to convert to a sequence.
+        Args:
+            sent (str): The input sentence to convert to a sequence.
+            symbol_list (str or List of string, optional): When the input is a string, available options include
```

Comment:
I think the optional in the docstring and optional in types means two different things (correct me if I am wrong).
Optional in types means that it can take in the value `None`. While optional in docstring means that it has default value.